include/sparrow_v01/layout/array_base.hpp

```diff
@@ -36,39 +38,88 @@ namespace sparrow
 
     protected:
 
-        array_base(arrow_proxy proxy);
+        array_base() = default;
         array_base(const array_base&) = default;
 
-        const arrow_proxy& data() const;
-
     private:
 
         virtual array_base* clone_impl() const = 0;
-
-        arrow_proxy m_proxy;
     };
 
     /**
-     * Base class for arrays with bitmap
+     * traits class that must be specialized by array
+     * classes inheriting from array_crtp_base.
      */
-    class array_with_bitmap : public array_base
+    template <class D>
+    struct array_inner_types;
+
+    /**
+     * Base class defining common interface for arrays.
+     *
+     * This class is a CRTP base class that defines and
+     * implements comme interface for arrays with a bitmap.
+     */
+    template <class D>
+    class array_crtp_base
     {
     public:
 
-        using bitmap_type = dynamic_bitset_view<const std::uint8_t>;
+        using size_type = std::size_t;
+        using difference_type = std::ptrdiff_t;
+        using bitmap_type = dynamic_bitset_view<std::uint8_t>;
```

Comment:
bitmap_type could also come from a type_traits class for the derived class D.
Since the bitmap_type will be the same for **most** layouts,we could have a base class for array_inner_types with some defaults

```Cpp
struct array_inner_types_base
{
    using bitmap_type = dynamic_bitset_view<std::uint8_t>;
};


template <typename T>
struct array_inner_types<primitive_array<T>> : array_inner_types_base
{
    // ...
    /// bitmap not needed
};

template <>
struct array_inner_types<union_array> : array_inner_types_base
{
    using bitmap_type = the_special_union_bitmap_type;
}
```

Reply:
I like this idea!